pkg/compare/compare.go

```diff
@@ -433,6 +443,14 @@ func (o *Options) Run() error {
 		return fmt.Errorf("failed to collect resources: %w", err)
 	}
 	r.IgnoreErrors(func(err error) bool {
+		if strings.Contains(err.Error(), "Object 'Kind' is missing") {
+			klog.Warningf(skipInvalidResources, extractPath(err.Error(), 3), "'Kind' is missing")
+			return true
+		}
```

Comment:
Instead of checking the error string we should check for the error type with As or better just use the helper function from apimachinery? https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/error.go#L93-L101

Same for the other check below

Reply:
I wish i could,
the function from apimachinery doesn't work and also errors,As does not work, the type of the return error is:
errors.errorString so cant check for the error type
and the string is:
unable to decode "testdata/InvalidResourcesAreSkipped/resources/d1.json": Object 'Kind' is missing in '{
  "glossary": {
    "title": "example glossary"
      }
}'

Reply:
Ah gotcha!